packages/syft/src/syft/oblv/deployment_client.py

```diff
@@ -221,6 +228,65 @@ def sanity_check_oblv_response(self, req: requests.Response):
                 f"Failed to perform the operation  with status {req.status_code}"
             )
 
+    def request_code_execution(
+        self,
+        inputs: Dict[AbstractNodeClient, Dict],
+        code: Union[str, Callable],
+        outputs: Iterable[str],
+    ):
+        if not inspect.isfunction(code) and not isinstance(code, str):
+            raise Exception("The code should either be a function or  string ...")
+
+        if inspect.isfunction(code):
+            code = inspect.getsource(code)
+
+        # Step1 : Restructing inputs
+        node_input_map = {}
+        clients = inputs.keys()
+        # TODO: 🟣 Simplify code structure for inputs fields
+        # TODO: 🟣 Additional sanity checks for inputs fields
+        for client, client_input_map in inputs.items():
+            if not isinstance(client, AbstractNodeClient):
```

Comment:
MAYBE: Could we move the validation to another function? (also the one from 255)?

In this way we keep separate the validation logic and the client mapping logic.